odoo/addons/base/models/res_device.py

```diff
@@ -34,6 +34,13 @@ class ResDeviceLog(models.Model):
     is_current = fields.Boolean("Current Device", compute="_compute_is_current")
     linked_ip_addresses = fields.Text("Linked IP address", compute="_compute_linked_ip_addresses")
 
+    def init(self):
+        self.env.cr.execute(SQL("""
+            CREATE INDEX ON %s (user_id, session_identifier, platform, browser) WHERE revoked = False
```

Comment:
You should name the index. But it seems like the index is not used in the queries you provided. `res_device_log__session_identifier_index` was used instead.
If you include last_activity and id, then maybe it will just access the index without going to the table.

Reply:
I applied your suggestion to add `last_activity` and `id` in the index.
We can see from the query plan that this is the index used
```
Index Only Scan using res_device_log__composite_idx
```
(The query plans in the commit message have been updated)